docs/metrics/iou.md

```diff
@@ -0,0 +1,170 @@
+
+$$
+\text{area} \left( \text{A} \cup \text{B} \right) = \text{area} \left( \text{A} \right) + \text{area} \left( \text{B} \right) - \text{area} \left( \text{A} \cap \text{B} \right)
+$$
+
+#### Examples: IoU of 2D Bounding Boxes
+The following examples show what IoU values look like in different scenarios with 2D bounding boxes:
+
+**Example 1: overlapping bounding boxes**
+
+![An example of overlapping bounding boxes](../assets/images/metrics-iou-example1.png)
+
+$$
+\begin{align}
+\text{IoU} \left( \text{A}, \, \text{B} \right)
+&= \frac {(10 - 5) \times (10 - 5)} {10 \times 10 + (15 - 5) \times (15 - 5) - (10 - 5) \times (10 - 5)} \\[1em]
+&= \frac {25} {100 + 100 - 25} \\[1em]
+&= 0.143
+\end{align}
+$$
+
+**Example 2: non-overlapping bounding boxes**
+
+![An example of non-overlapping bounding boxes](../assets/images/metrics-iou-example2.png)
+
+$$
+\begin{align}
+\text{IoU} \left( \text{A}, \, \text{B} \right)
+&= \frac {0} {10 \times 10 + (15 - 10) \times (15 - 10) - 0} \\[1em]
+&= \frac {0} {100 + 25 - 0} \\[1em]
+&= 0.0
+\end{align}
+$$
+
+
+**Example 3: completely matching bounding boxes**
+
+![An example of completely matching bounding boxes](../assets/images/metrics-iou-example3.png)
+
+$$
+\begin{align}
+\text{IoU} \left( \text{A}, \, \text{B} \right)
+&= \frac {10 \times 10} {10 \times 10 + 10 \times 10 - 10 \times 10} \\[1em]
+&= \frac {100} {100 + 100 - 100} \\[1em]
+&= 1.0
+\end{align}
+$$
+
+
+### Segmentation Mask
+
+**A segmentation mask** is a 2D image where each pixel is a class label commonly used in semantic segmentation tasks. The prediction shape matches the ground truth shape (width and height), with a channel depth equivalent to the number of class labels to be predicted. Each channel is a binary mask that labels areas where a specific class is present:
+
+![An example of segmentation mask](../assets/images/metrics-iou-seg-mask.png)
+<p style="text-align: center; color: gray;">
+    From left to right: the original RGB image, the ground truth segmentation mask, and the prediction segmentation mask
+</p>
+
+
+The IoU metric measures the **intersection** (the number of pixels **common** between the ground truth and prediction masks, **true positive (TP)**) divided by the **union** (the **total** number of pixels present across **both** masks, **TP + false negative (FN) + false positive (FP)**). And here is the formula for the IoU metric for a segmentation mask:
+
+$$
+\text{IoU} \left( \text{A}, \, \text{B} \right) = \frac {\text{TP}} {\text{TP} + \text{FN} + \text{FP}}
+$$
+
+Let’s look at what TP, FP, and FN look like on a segmentation mask:
+
+![An example of segmentation mask with results](../assets/images/metrics-iou-seg-mask-results.png)
```

Comment:
This image is great. What have you been using to draw these?

Reply:
There is this app on ipad called Freeform.  The mask is hand-drawn though. 😂